### PR TITLE
[bitnami/kube-prometheus] feat: :lock: Enable networkPolicy

### PR DESCRIPTION
**Pull request by javsalgar**
Signed-off-by: Javier Salmeron Garcia <jsalmeron@vmware.com>

### Description of the change

This PR normalizes the use of NetworkPolicy in the chart. Adds all Bitnami standards for NetworkPolicies as well as enabling it by default, in order to comply with security checklists.

### Benefits

More security in the chart

### Applicable issues

- fixes #

### Checklist

- [ ] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/kube-prometheus/templates/alertmanager/networkpolicy.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    {{- if .Values.alertmanager.networkPolicy.extraEgress }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.alertmanager.networkPolicy.extraEgress "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- end }}
+  ingress:
+    {{/* TODO: Allow configuring alertmanager container ports */}}
+    - ports:
+        - port: 8080
+        - port: 9093
+        - port: 9094
```
Comment (juan131) [inline bitnami/kube-prometheus/templates/alertmanager/networkpolicy.yaml:37]: Shouldn't these values be set based on `.Values.alertmanager.containerPorts.*`?

File: bitnami/kube-prometheus/templates/prometheus-operator/networkpolicy.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  {{- else }}
+  egress:
+    {{- if .Values.operator.networkPolicy.extraEgress }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.operator.networkPolicy.extraEgress "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- end }}
+  ingress:
+    {{/* TODO: Allow configuring prometheus-operator container ports */}}
+    - ports:
+        - port: 8080
```
Comment (juan131) [inline bitnami/kube-prometheus/templates/prometheus-operator/networkpolicy.yaml:35]: Same here, see TODO

File: bitnami/kube-prometheus/templates/prometheus/networkpolicy.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  {{- end }}
+  ingress:
+    {{/* TODO: Allow configuring prometheus container ports */}}
+    - ports:
+        - port: 9090
+        - port: 8080
+        {{- if .Values.prometheus.thanos.create }}
+        - port: 10901
+        - port: 10902
+        {{- end }}
```
Comment (juan131) [inline bitnami/kube-prometheus/templates/prometheus/networkpolicy.yaml:40]: same here

**Review by juan131 — CHANGES_REQUESTED** [review]

File: bitnami/kube-prometheus/templates/alertmanager/networkpolicy.yaml (hunk 1 above)
Comment (javsalgar) [reply bitnami/kube-prometheus/templates/alertmanager/networkpolicy.yaml:38]: They are hardcoded in the operator. Comment updated

File: bitnami/kube-prometheus/templates/prometheus/networkpolicy.yaml (hunk 3 above)
Comment (javsalgar) [reply bitnami/kube-prometheus/templates/prometheus/networkpolicy.yaml]: Prometheus ports are hardcoded in the operator. Added containerPorts for thanos

**Pull request merged**